### ISSUE TITLE
Classifiers are very bad?

### ISSUE THREAD

**Issue by YLaCoon** [cc-hpc-itwm/DeepFakeDetection#7]
You report accuracy on unbalanced data splits. Why? Accuracy is not good for unbalanced data. Please update the results with `roc_auc_score`! For example a simple all 0 prediction is better than your classifiers. I only changed one line "SIMPLE". This "amazing" model gets **0.9340659340659341** accuracy ...

```
import numpy as np
from sklearn.svm import SVC
from sklearn.linear_model import LogisticRegression
import pickle

#train
pkl_file = open('train_3200.pkl', 'rb')
data = pickle.load(pkl_file)
pkl_file.close()
X = data["data"]
y = data["label"]

svclassifier_r = SVC(C=6.37, kernel='rbf', gamma=0.86)
svclassifier_r.fit(X, y)
logreg = LogisticRegression(solver='liblinear', max_iter=1000)
logreg.fit(X, y)

#test
pkl_file = open('test_full.pkl', 'rb')
data = pickle.load(pkl_file)
pkl_file.close()
X_ = data["data"]
y_ = data["label"]

SVM = svclassifier_r.score(X_, y_)
LR = logreg.score(X_, y_)
print("SVM: "+str(SVM))
print("LR: "+str(LR))
print("SIMPLE:"+str((np.zeros_like(y_) == y_).mean()))
```

Please let me know if I missed something. I look forward to your response.

**Comment by YLaCoon** [comment]
The same for video level predictions... Your split is unbalanced. 92.5% of the videos are real. Your classifier achieves `0.895464` accuracy. What's the point in providing these metrics???

**Comment by MARD1NO** [comment]
> The same for video level predictions... Your split is unbalanced. 92.5% of the videos are real. Your classifier achieves `0.895464` accuracy. What's the point in providing these metrics???

Hi,I also run this code.I also notice the data in train data("train_3200.pkl") is right.real and fake data is  
1 : 1.
but in test data I notice that label'0' : label'1' is 17000:1200
Otherwise i also wonder why the figure in paper, like Figure1, its PowerSpectrum is 0.0~1.0.
but in Figure 10,its PowerSpectrum is around 100~300+?

**Comment by MARD1NO** [comment]
> The same for video level predictions... Your split is unbalanced. 92.5% of the videos are real. Your classifier achieves `0.895464` accuracy. What's the point in providing these metrics???

Hey,now i use train_test_split in the train dataset.Actually the accuracy is 89% better than all zero accuracy 51.1%

**Comment by YLaCoon** [comment]
You don't show your code... Note that you introduce a leak if you naively split the data. This is because almost the same faces appear in the train and test set. That's why i asked for the train/val/test split in another issue.

**Comment by RicardDurall** [comment]
As already mentioned, the training data is balanced and this is the important point.
The reason why the testing is so unbalanced is due to the amount of real and fake images that FaceForensics++ provides. This dataset contains many more fake than real examples. That's why if you just classified everything as a fake you would end up having really high score. However, our implementation does not do that.
Finally, we will provide the input images, so that you can create your own split.

**Comment by YLaCoon** [comment]
> As already mentioned, the training data is balanced and this is the important point.

I disagree. The important point is the **test** performance. The chosen metric "accuracy" is not able measure the performance as the test data is unbalanced. I thus proposed that you switch to `roc_auc_score` as metric! Using it is as simple as `from sklearn.metrics import roc_auc_score`. I look forward to your updates and scores given this **meaningful** metric for unbalanced data!
